File: entrenamiento/models/flechas.py
# -*- coding: utf-8 -*-

from entrenamiento.app.app import db
from entrenamiento.models.base import BaseModel

class Flechas(BaseModel):
    ''' Tiene toda la informacion de las flechas que puede llegar a usar
    el arquero en cuestion.

    :param int id: un identificador autogenerado.

    :param int id_usuario: el identificador del usuario a quien le pertence toda
                           la informacion de las flechas.

    :param str comentario: un comentario sobre las flechas que uno esta usando
                           Por ejemplo: "las corte 2mm de atras, etc.."

    :param int id_tipo_uso: identifica si las flechas se las usa solo para indoor,
                            o solo para outdoor o para ambas.

    :param int id_marca_flechas: el identificador de la marca de las flechas (easton,
                                 cartel, etc..)

    :param int id_modelo_flechas: el identificador del modelo de flechas (carbon one,
                                  jazz, etc...)

    :param int calibre_flechas: indica el calibre de la flecha. Para las jazz, y
                                las platinum el mismo va a ser algo como 1916, pero
                                para las flechas de carbono va a ser algo como 600

    :param int punta: informacion de la punta. Para flechas de outdoor indica el peso
                      en grames de las mismas.

    :param float largo: el largo de las flechas que usa.

    :param str info_nock: toda la informacion del nock (marca, modelo, tamaño)

    :param str info_timones: toda la informacion sobre los timones que se estan usando
                             (color, largo, marca y modelo, etc..)
    '''

    id = db.Column(db.Integer, primary_key=True, nullable=False)
    id_usuario = db.Column(db.Integer,
                           db.ForeignKey('usuario.id', ondelete='CASCADE'),
                           nullable=False)
    comentario = db.Column(db.Text)

    id_tipo_uso = db.Column(db.Integer,
                            db.ForeignKey('tipo_uso_flechas.id'),
                            nullable=False)

    id_marca_flechas = db.Column(db.Integer,
                                 db.ForeignKey('marca_flechas.id'),
                                 nullable=False)

    id_modelo_flechas = db.Column(db.Integer,
                                  db.ForeignKey('modelo_flechas.id'),
                                  nullable=False)

    calibre_flechas = db.Column(db.Integer)
    punta = db.Column(db.Integer)
    largo = db.Column(db.Float)
    info_nock = db.Column(db.String(1024))
    info_timones = db.Column(db.String(1024))

File: entrenamiento/static/js/apps/crud/flechas.js
/**
 * La applicacion que se encarga de todo el tema del CRUD
 * de pagos
 */
var FlechasCrudApplication = FieldCrudApp.$extend({

    __init__: function(element, historyManager, apiManager, databaseInformation) {
        var formFields = [
            new FormFieldData('id_tipo_uso', null),
            new FormFieldData('id_marca_flechas', null),
            new FormFieldData('id_modelo_flechas', null),
            new FormFieldData('calibre_flechas', 'En el caso de las flechas de carbono es un numero (660). En el caso de las jazz son dos numeros juntos (1913)'),
            new FormFieldData('punta', 'El peso de la punta en grames. Si es una punta de jazz o Plantinum, entonces es el calibre de la punta (1913)'),
            new FormFieldData('largo', 'El largo de las flecha en pulgadas'),
            new FormFieldData('info_nock', 'Toda la informacion del nock. Por ejemplo: Beiter IN-OUT 2x95-S1 Verde'),
            new FormFieldData('info_timones', 'Toda la informacion de los timones. Por ejemplo: Kurly Vanes Verdes Homologados 45mm'),
        ];
        this.$super(element, historyManager, apiManager, databaseInformation,
                    'flechas',
                    ['id_usuario', 'id_marca_flechas', 'id_modelo_flechas'],
                    formFields,
                    null);
    }
});

